passage/src/main/java/id/passage/android/Passage.kt

```diff
@@ -1,759 +1,66 @@
-            } catch (e: java.lang.Exception) {
-                throw FinishSocialAuthenticationException.convert(e)
-            }
-        PassageSocial.verifier = ""
-        handleAuthResult(authResult)
-        return authResult
-    }
-
-    // endregion
-
-    // region USER METHODS
-
-    /**
-     * Get Current User
-     *
-     * Returns an instance of PassageUser, which represents an authenticated Passage user.
-     * The PassageUser class has methods that can be used to retrieve data on the current user
-     * which require authentication.
-     * @return PassageUser?
-     */
-    public suspend fun getCurrentUser(): PassageUser? {
-        if (isUsingTokenStore && ApiClient.accessToken == null) {
-            tokenStore.attemptRefreshTokenStore()
-        }
-        val user =
-            try {
-                PassageUser.getCurrentUser()
-            } catch (e: Exception) {
-                Log.w(TAG, "Getting current user failed. ${e.message ?: e.toString()}")
-                null
-            }
-        return user
-    }
-
-    /**
-     * Sign Out Current User
-     *
-     * If using Passage Token Store, calling this method will revoke the user's refresh token,
-     * clear all tokens from the Passage Token Store, and set the client's access token to null.
-     * If not using Passage Token Store, calling this method will set the client's access token to null.
-     * @return void
-     * @throws PassageTokenException
-     */
-    public suspend fun signOutCurrentUser() {
-        if (isUsingTokenStore) {
-            tokenStore.clearAndRevokeTokens()
-        } else {
-            setAuthToken(null)
-        }
-    }
-
-    /**
-     * Identifier Exists
-     *
-     * Checks if the identifier provided exists for the application. This method should be used to
-     * determine whether to register or log in a user. This method also checks that the app supports
-     * the identifier types (e.g., it will throw an error if a phone number is supplied to an app
-     * that only supports emails as an identifier).
-     * @return PassageUser?
-     */
-    public suspend fun identifierExists(identifier: String): PassageUser? {
-        val usersAPI = UsersAPI(BASE_PATH, passageClient)
-        val modelsUser =
-            try {
-                usersAPI.checkUserIdentifier(appId, identifier).user
-            } catch (e: Exception) {
-                return null
-            } ?: return null
-        return PassageUser.convertToPassageUser(modelsUser)
-    }
-
-    // endregion
-
-    // region OVERRIDE METHODS
     public fun overrideBasePath(newPath: String) {
         BASE_PATH = newPath
     }
-    // endregion
-
-    // region PRIVATE METHODS
-
-    /**
-     * Handle Auth Result
-     *
-     * Anytime a user gets a new Auth Result, update the tokens in Passage Token Store (if applicable).
-     * @param authResult The new Auth Result
-     */
-    private fun handleAuthResult(authResult: PassageAuthResult?) {
```

Comment:
This method handled saving tokens to device, which we still need to do. How will we do this now?

Reply:
For this function, we don't need the first part (usingTokenStore) anymore. We're handling tokens ourselves now. Also, all the classes depend on tokenStore, and they each call "tokenStore.setTokens(authResult)" on their own.